frontend/src/metabase/visualizations/lib/settings/series.js

```diff
@@ -154,19 +155,28 @@ export function seriesSetting({ readDependencies = [], def } = {}) {
     "line.missing": {
       title: t`Replace missing values with`,
       widget: "select",
-      props: {
+      getProps: (_series, settings, _onChange, { settings: vizSettings }) => ({
         options: [
           { name: t`Zero`, value: "zero" },
           { name: t`Nothing`, value: "none" },
-          { name: t`Linear Interpolated`, value: "interpolate" },
+          ...(hasInterpolatedLineMissingOption(
+            settings["display"],
+            vizSettings["stackable._default_stack_type"],
+          )
+            ? [{ name: t`Linear Interpolated`, value: "interpolate" }]
+            : []),
         ],
-      },
+      }),
       getHidden: (single, settings) =>
         !LINE_DISPLAY_TYPES.has(settings["display"]),
       getDefault: (single, settings, { settings: vizSettings }) =>
         // use legacy global line.missing setting if present
-        getSeriesDefaultLineMissing(vizSettings),
-      readDependencies: ["display"],
+        getSeriesDefaultLineMissing(
+          vizSettings,
+          settings["display"],
+          vizSettings["stackable._default_stack_type"],
```

Comment:
Instead of passing just `vizSettings` I think deliberatly pass more props makes it clear that this is the dependencies. I could call these `settings` under the hood, but it wouldn't be clear where these settings are used.